draft-ietf-tls-esni.md

```diff
@@ -409,12 +409,18 @@ ClientHelloOuter, and a digest of the complete ClientHelloInner.
 When sending ClientHello, the client first computes ClientHelloInner, including
 any PSK binders, and then MAY substitute extensions which it knows will be
 duplicated in ClientHelloOuter. To do so, the client computes a hash of the
-entire ClientHelloInner message with the same hash as for the KDF used to
-encrypt ClientHelloInner. Then, the client removes and replaces extensions from
-ClientHelloInner with a single "outer_extensions" extension. The list of
-`outer_extensions` include those which were removed from ClientHelloInner, in
-the order in which they were removed. The hash contains the full
-ClientHelloInner hash computed above.
+entire ClientHelloInner message as:
+
+~~~
+  hash = Expand(Extract("", inner), "ech_inner_digest", Nh)
```

Comment:
Let's make a note to change this to something other than `hash`, too. (If we keep it!)

Reply:
`config_id` ? or, for now: 

```suggestion
  config_digest = Expand(Extract("", inner), "ech_inner_digest", Nh)
```

Reply:
Well this is the hash of the inner CH, not of the config. Let's just change this name in a separate PR.

Reply:
Agreed.